marimo/_runtime/runtime.py

```diff
@@ -1088,40 +1092,38 @@ async def run(
                 filtered_requests.append(request)
 
         await self._run_cells(
-            self.mutate_graph(
-                filtered_requests,
-                deletion_requests=[
-                    # Always delete the scratchpad cell
-                    DeleteCellRequest(cell_id=SCRATCH_CELL_ID)
-                ],
-            )
+            self.mutate_graph(filtered_requests, deletion_requests=[])
         )
 
-    async def run_scratchpad(self, request: ExecuteScratchpadRequest) -> None:
-        LOGGER.warning("Running scratchpad cell")
+    async def run_scratchpad(self, code: str) -> None:
         roots = {SCRATCH_CELL_ID}
-        # Mutated graph by adding scratchpad cell
-        self.mutate_graph(
-            execution_requests=[
-                ExecutionRequest(SCRATCH_CELL_ID, request.code)
-            ],
-            deletion_requests=[],
-        )
+
+        # If cannot compile, don't run
+        try:
+            cell = compile_cell(code, SCRATCH_CELL_ID)
+            cell.defs.clear()  # remove definitions
+            cell.refs.clear()  # remove definitions
+        except SyntaxError:
+            return
```

Comment:
Shouldn't this broadcast broadcast error before return?
For dry, you can breakout the compilation and error handling from `_try_registering_cell` then just `CellOp.broadcast_error`